lib/cmock_header_parser.rb

```diff
@@ -73,28 +73,79 @@ def remove_nested_pairs_of_braces(source)
     return source
   end
 
+  # Return the number of pairs of braces/square brackets in the function provided by the user
+  # +source+:: String containing the function to be processed
+  def count_number_of_pairs_of_braces_in_function(source)
+    is_function_start_found = false
+    curr_level = 0
+    total_levels = 0
+
+    source.each_char do |c|
+      if ("{" == c)
+        curr_level += 1
+        total_levels  +=1
```

Comment:
Based on the way you are using it, I think this is functionally ok... but total_levels isn't counting levels so much as it's counting pairs. For example, the following would have 3 total levels, even though it's only two deep:

```
{
  { }
  { }
}
```

Maybe this should be renamed to `total_pairs`?

Reply:
Yes, agreed, total_pairs makes more sense